File: presentation/src/main/java/com/khalturin/dmitriy/presentation/di/component/PresenterComponent.java
package com.khalturin.dmitriy.presentation.di.component;

import com.khalturin.dmitriy.presentation.di.PerPresenter;
import com.khalturin.dmitriy.presentation.di.module.PresenterModule;
import com.khalturin.dmitriy.presentation.view.activity.BookmarksActivity;
import com.khalturin.dmitriy.presentation.view.activity.FeedActivity;
import com.khalturin.dmitriy.presentation.view.activity.NewsActivity;

import dagger.Subcomponent;

/**
 * Created by Dmitriy Khalturin <dmitry.halturin.86@gmail.com>
 * for android-rss-reader on 10.04.18 1:10.
 */

@PerPresenter
@Subcomponent(modules = {
  PresenterModule.class
})
public interface PresenterComponent {

  void inject(FeedActivity activity);
  void inject(NewsActivity activity);
  void inject(BookmarksActivity activity);

}

File: rssreader/src/main/kotlin/dmitriykhalturin/rssreader/di/component/PresenterComponent.kt
package dmitriykhalturin.rssreader.di.component

import dmitriykhalturin.rssreader.di.PerPresenter
import dmitriykhalturin.rssreader.di.module.PresenterModule
import dmitriykhalturin.rssreader.view.activity.ArticleActivity
import dmitriykhalturin.rssreader.view.activity.BookmarksActivity
import dmitriykhalturin.rssreader.view.activity.FeedActivity
import dagger.Component

/**
 * Created by Dmitriy Khalturin <dmitry.halturin.86@gmail.com>
 * for android-rss-reader on 26.10.18 21:45.
 */
@PerPresenter
@Component(
  modules = [
    PresenterModule::class
  ]
)
interface PresenterComponent {

  fun inject(activity: FeedActivity)
  fun inject(activity: ArticleActivity)
  fun inject(activity: BookmarksActivity)

}
